### ISSUE TITLE
Disable plugin options requiring a missing dependency

### ISSUE THREAD

**Issue by Kraymer** [beetbox/beets#1963]
When a library related to a plugin option (langdetect for translate, bs4 for google lyrics backend, etc) is missing, it should be detected at the plugin init and the offending option should be disabled with a message printed once pointing to the docs.
Currently we have an error/warning for each song.

**Comment by jackwilsdon** [comment]
It would be cool if plugins themselves could tell beets core about the issue and beets would disable the plugin automatically. I wonder if this is possible?

**Comment by GuilhermeHideki** [comment]
maybe with `__import__ / importlib` help. You would try to import and logs a message if can't import something.

**Comment by wordofglass** [comment]
This is exactly what beets already does (see https://github.com/beetbox/beets/blob/master/beets/plugins.py#L267 which also catches any `ImportError`. Raising any error during plugin init will effectively disable the plugin). The problem with lyrics is different, though. The lack of either of these modules should not shutdown the whole plugin, but only deactivate the corresponding sources. The recent patch does that, but it will also print a warning every time lyrics are fetched, which messes up the UI during imports. That's what Kraymer was about, the plugin would need some refactoring to print that warning only once.

**Comment by GuilhermeHideki** [comment]
adding something like this at the top would work / is too hacky?
```py
try:
  import langdetect
except ImportError:
  disable_source('bing')
  warn_user()
```

**Comment by wordofglass** [comment]
The warning should not print when the source is not enabled through the config, though, as that combination may aswell be intentional. Have a look at `LyricsPlugin.__init__`, the google source does that when no API key is given.

**Comment by wordofglass** [comment]
Fixed in #1968
